autoprof/fit/base.py

```diff
@@ -20,87 +20,106 @@ class BaseOptimizer(object):
         initial_state: optional initialization for the parameters as a 1D tensor [tensor]
         max_iter: maximum allowed number of iterations [int]
         relative_tolerance: tolerance for counting success steps as: 0 < (Chi2^2 - Chi1^2)/Chi1^2 < tol [float]
-    
+
     """
-    def __init__(self, model: 'Autorof_Model', initial_state: Sequence = None, relative_tolerance: float = 1e-3, **kwargs) -> None:
+
+    def __init__(
+        self,
+        model: "Autorof_Model",
+        initial_state: Sequence = None,
+        relative_tolerance: float = 1e-3,
+        **kwargs,
+    ) -> None:
+        """
+        Initializes a new instance of the class.
+
+        Args:
+            model (object): An object representing the model.
+            initial_state (Union[None, Tensor]): The initial state of the model could be any tensor.
```

Comment:
The type for `initial_state` can be any sequence so it would be `Optional[Sequence]`